lib/application.js

```diff
@@ -208,7 +208,7 @@ module.exports = class Application extends Emitter {
     request.response = response
     response.request = request
     context.originalUrl = request.originalUrl = req.url
-    context.state = {}
+    context.state = Object.assign({}, context.state)
```

Comment:
It might be better to use `Object.create()` here, to inherit the prototype from `context.state`.

```suggestion
    context.state = Object.create(context.state)
```

Reply:
Sorry, just saw this PR fail because my suggestion broke it. This should fix it:

```suggestion
    context.state = Object.create(context.state || {})
```

This would be a great place to use [logical nullish assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment), but Koa has to support Node v12 which doesn't support that, and I don't think Koa transpiles its code (?), so better safe than sorry.

Reply:
Tried this one locally but some tests are faiiling, still does not work

Reply:
Huh okay. Sorry for the detour. This still definitely covers the use case. Thanks for tackling this PR!